### PR TITLE
minimize code duplication

### PR DESCRIPTION
**Pull request by alex-parham1**

### CODE REVIEW

File: ArcanicVosforator/src/main/java/org/example/arcanicvosforator/ArcanicVosforatorInterface.java [hunk 1]
```diff
@@ -82,240 +82,85 @@ public void start(Stage stage) {
...
-                quillsArcaneSetFilter.setSelected(false);
-                necraliskArcaneSetFilter.setSelected(false);
-                duviriArcaneSetFilter.setSelected(false);
-                caviaArcaneSetFilter.setSelected(false);
-                luaArcaneSetFilter.setSelected(false);
-                fortunaArcaneSetFilter.setSelected(false);
-                mirrorDefenseArcaneSetFilter.setSelected(false);
-                plagueStarArcaneSetFilter.setSelected(false);
-                voxSolarisArcaneSetFilter.setSelected(false);
+                clearSelectedFilter("eidolon");
```
Comment (alex-parham1) [inline ArcanicVosforator/src/main/java/org/example/arcanicvosforator/ArcanicVosforatorInterface.java:85]: made a method to reduce the large amount of duplicated code here.

File: ArcanicVosforator/src/main/java/org/example/arcanicvosforator/ArcanicVosforatorInterface.java [hunk 2]
```diff
@@ -378,558 +223,426 @@ private void selectArcaneEvent() {
             switch (arcane) {
                 // arbitration arcane set.
                 case "Arcane Blade Charger":
-                    getData().addToTotalVosforCount(getData().getVosforRates().get(4), quantity);
-                    getData().addToGrandTotalVosforCount();
+                    addToVosforCount(4, quantity);
```
Comment (alex-parham1) [inline ArcanicVosforator/src/main/java/org/example/arcanicvosforator/ArcanicVosforatorInterface.java:226]: made a method to reduce the code duplication here

File: ArcanicVosforator/src/main/java/org/example/arcanicvosforator/ArcanicVosforatorInterface.java [hunk 3]
```diff
@@ -941,27 +654,67 @@ private void selectArcaneEvent() {
...
-        caviaArcaneSetFilter.setSelected(false);
-        luaArcaneSetFilter.setSelected(false);
-        fortunaArcaneSetFilter.setSelected(false);
-        mirrorDefenseArcaneSetFilter.setSelected(false);
-        plagueStarArcaneSetFilter.setSelected(false);
-        voxSolarisArcaneSetFilter.setSelected(false);
+        clearSelected();
+    }
+
+    private void clearSelectedFilter(String filter){
```
Comment (alex-parham1) [inline ArcanicVosforator/src/main/java/org/example/arcanicvosforator/ArcanicVosforatorInterface.java:672]: this method is indeed technically slower than the previous method but only by a few line executions
upside being the code blocks are much easier to visually digest

File: ArcanicVosforator/src/main/java/org/example/arcanicvosforator/ArcanicVosforatorInterface.java [hunk 4]
```diff
@@ -941,27 +654,67 @@ private void selectArcaneEvent() {
...
-        quillsArcaneSetFilter.setSelected(false);
-        necraliskArcaneSetFilter.setSelected(false);
-        duviriArcaneSetFilter.setSelected(false);
-        caviaArcaneSetFilter.setSelected(false);
-        luaArcaneSetFilter.setSelected(false);
-        fortunaArcaneSetFilter.setSelected(false);
-        mirrorDefenseArcaneSetFilter.setSelected(false);
-        plagueStarArcaneSetFilter.setSelected(false);
-        voxSolarisArcaneSetFilter.setSelected(false);
+        clearSelected();
```
Comment (alex-parham1) [inline ArcanicVosforator/src/main/java/org/example/arcanicvosforator/ArcanicVosforatorInterface.java:669]: you can take or leeave this one. just really quickly popped this one in here too.

File: ArcanicVosforator/src/main/java/org/example/arcanicvosforator/ArcanicVosforatorInterface.java [hunk 5]
```diff
@@ -378,558 +223,426 @@ private void selectArcaneEvent() {
...
                     break;
                 case "Emergence Dissipate":
-                    getData().addToTotalVosforCount(getData().getVosforRates().get(6), quantity);
-                    getData().addToGrandTotalVosforCount();
+                    addToVosforCount(6, quantity);
                     break;
                 case "Molt Augmented":
-                    getData().addToTotalVosforCount(getData().getVosforRates().get(6), quantity);
-                    getData().addToGrandTotalVosforCount();
+                    addToVosforCount(6, quantity);
```
Comment (alex-parham1) [inline ArcanicVosforator/src/main/java/org/example/arcanicvosforator/ArcanicVosforatorInterface.java:645]: small method to minimize the code reuse and improve readablity of the switch.
logic is truly unchaged, just minimizing reused code.

also makes it easier to expand in the future with more aracane additions.

File: ArcanicVosforator/src/main/java/org/example/arcanicvosforator/ArcanicVosforatorInterface.java (hunk 3 above)
Comment (RushoBinnabi) [reply ArcanicVosforator/src/main/java/org/example/arcanicvosforator/ArcanicVosforatorInterface.java:672]: :)

**Pull request merged**